Backend/app/dashboard/CommonAPIs/src/Dashboard/RiderPlatform/Ride.hs

```diff
@@ -0,0 +1,84 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingVia #-}
+
+module Dashboard.RiderPlatform.Ride
+  ( module Dashboard.RiderPlatform.Ride,
+    module Reexport,
+  )
+where
+
+import Dashboard.Common as Reexport
+import qualified Dashboard.Common as DP
+import Kernel.External.Maps
+import Kernel.Prelude
+import Kernel.Types.Centesimal
+import Kernel.Types.Id
+import Servant
+
+---------------------------------------------------------
+-- share ride info--------------------------------------
+
+type ShareRideInfoAPI =
+  Capture "rideId" (Id DP.Ride)
+    :> "share"
+    :> Get '[JSON] ShareRideInfoRes
+
+data ShareRideInfoRes = ShareRideInfoRes -- handle repetition
+  { id :: Id Ride,
+    bookingId :: Id Booking,
+    status :: RideStatus,
+    driverName :: Text,
+    driverRating :: Maybe Centesimal,
+    driverMobileNumber :: Text,
+    vehicleNumber :: Text,
+    vehicleModel :: Text,
+    vehicleColor :: Text,
+    trackingUrl :: Maybe BaseUrl,
+    rideStartTime :: Maybe UTCTime,
+    rideEndTime :: Maybe UTCTime,
+    userFirstName :: Maybe Text,
+    userLastName :: Maybe Text,
+    fromLocation :: BookingLocation,
+    toLocation :: Maybe BookingLocation
+  }
+  deriving (Generic, Show, ToSchema, FromJSON, ToJSON)
+
+data RideStatus
+  = NEW
+  | INPROGRESS
+  | COMPLETED
+  | CANCELLED
+  deriving (Show, Eq, Ord, Read, Generic, ToJSON, FromJSON, ToSchema)
+
+data BookingLocation = BookingLocation
```

Comment:
Is there any difference between already existing one and this ?

Reply:
@RitikaHanish It is not present on dashboard side. @prashant601 Can you check if it is feasible to import these types from rider/driver app?